src/game-context.js

```diff
@@ -0,0 +1,84 @@
+import { useContext, createContext, useState, useEffect } from "react";
+import { dataarray } from "./dataarray";
+const gameContext = createContext();
+// dataarray.sort(() => Math.random() - 0.5);
+const GameProvider = ({ children }) => {
+  const [status, setstatus] = useState("Let's Go!");
+  const [flipCard, setFlipCard] = useState(dataarray);
+  const [twoCounter, setTwoCounter] = useState(0);
+  const [turnCounter, setTurnCounter] = useState(0);
+  const matchNotFoundHandler = (objsToCheck) => {
+    console.log("not matching");
+    setstatus("doesnt match :(");
+    setTurnCounter((prev) => prev + 1);
+    setTwoCounter(0);
+    setTimeout(() => {
+      setFlipCard((prev) =>
+        prev.map((obj) =>
+          obj.letter.toUpperCase() === objsToCheck[0].letter.toUpperCase() ||
+          obj.letter.toUpperCase() === objsToCheck[1].letter.toUpperCase()
```

Comment:
I don't seem to understand why have you use `toUpperCase` at plenty of places.
If anyways you're showing all letters in upper case in the UI, then why have you put some letter as lower case in the data?

Reply:
Here, the data has two letters repeating but their case is different. The logic is sort of hacky from the array of data itself, hence the to uppercases.